### ISSUE TITLE
Rails/HasManyOrHasOneDependent incorrectly flags `with_options` blocks

### ISSUE THREAD

**Issue by rdunlop** [bbatsov/rubocop#4747]
My code has:

```ruby
class Registrant < ApplicationRecord
  with_options dependent: :destroy do
    has_one :contact_detail, autosave: true, inverse_of: :registrant
    has_one :standard_skill_routine
  end
end
```

And the rubocop HasManyOrHasOneDependent flags it:

```
app/models/registrant.rb:49:5: C: Rails/HasManyOrHasOneDependent: Specify a :dependent option.
    has_one :contact_detail, autosave: true, inverse_of: :registrant
    ^^^^^^^
app/models/registrant.rb:50:5: C: Rails/HasManyOrHasOneDependent: Specify a :dependent option.
    has_one :standard_skill_routine
    ^^^^^^^
```

This is unexpected, since I am specifying a `dependent` option for the has_one...just not directly (which is permitted by rails)

http://api.rubyonrails.org/classes/Object.html#method-i-with_options
--------

## Expected behavior

Not flag my code for having no `dependent:` option.

## Actual behavior

Flags the code as needing a `depndent:` option

## Steps to reproduce the problem

See above code

## RuboCop version

0.50.0

**Comment by bbatsov** [comment]
Yeah, guess we didn't factor this when the cop was developed. Should be relatively easy to fix.
